db.go

```diff
@@ -102,7 +105,7 @@ func (db *sqlDB) BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, err
 // The args are for any placeholder parameters in the query.
 // Exec uses the RW-database as the underlying db connection
 func (db *sqlDB) Exec(query string, args ...interface{}) (sql.Result, error) {
-	return db.ExecContext(context.Background(), query, args...)
+	return db.ExecContext(backgroundCtx, query, args...)
```

Comment:
This will make all queries use the same context, which internally use mutexes and channels to protect data structures, I think this may decrease performance under load.

Reply:
> Wait it contains a lot of changes in 1 PR?

Well you can merge my 1st #32  , i have already rebased then branch in this PR

Reply:
> This will make all queries use the same context, which internally use mutexes and channels to protect data structures, I think this may decrease performance under load.

@bxcodec @RangelReale look into the contents of context.Background(). Here is it. This is the code for ur ref

<img width="481" alt="image" src="https://github.com/bxcodec/dbresolver/assets/24226219/66dc8020-1c2a-4fd0-a46f-319c88005e3a">

As you can clearly see 

```
/ Background returns a non-nil, empty Context. It is **never canceled**, has no
// values, and has no **deadline**. It is typically used by the main function,
// initialization, and tests, and as the **top-level Context** for incoming
// requests.

```
Also its not susceptible to cascading terminations of child contexts...

Same applies to TODO

Reply:
@bxcodec the -race tests are passing with 10 Test cases running in parallel !. So the code is concurrent safe.

Earlier tests completion averaged at 8-9sec
Now tests are completing at 5-6sec

Reply:
If you see the [Go source code where db.Exec calls db.ExecPrepared](https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/database/sql/sql.go;l=1647), they call `context.Background()` every time, I think it is better to follow their practices.

Reply:
I don't understand that part either. CPU cycles are unnecessary wasted with redundant library calls.

Therefore I have raised a PR - https://github.com/golang/go/pull/60575